### PR TITLE
Fix batching to enable randomized game params + vmapping

### PR DESCRIPTION
**Pull request by StoneT2000**

### CODE REVIEW

File: src/luxai_s3/env.py [hunk 1]
```diff
@@ -9,79 +9,144 @@
...
+        For each team, generate a integer vision power array over the map.
         For each unit in team, add unit sensor range value (its kind of like the units sensing power/depth) to each tile the unit's sensor range
         Clamp the vision power array to range [0, unit_sensing_range].
 
         With 2 vision power maps, take the nebula vision mask * nebula vision power and subtract it from the vision power maps.
         Now any time the vision power map has value > 0, the team can sense the tile. This forms the sensor mask
         """
 
-        vision_power_map_padding = params.unit_sensor_range
+        vision_power_map_padding = self.fixed_env_params.unit_sensor_range
```
Comment (CJBoey) [inline src/luxai_s3/env.py:112]:
```suggestion
        max_sensor_range = env_params_ranges["unit_sensor_range"][-1]
        vision_power_map_padding = max_sensor_range
```

Comment (CJBoey) [inline src/luxai_s3/env.py:112]:
```suggestion
        max_sensor_range = env_params_ranges["unit_sensor_range"][-1]
        vision_power_map_padding = max_sensor_range
```

File: src/luxai_s3/env.py [hunk 2]
```diff
@@ -9,79 +9,144 @@
...
+            shape=(
+                self.fixed_env_params.num_teams,
+                self.fixed_env_params.map_height + 2 * vision_power_map_padding,
+                self.fixed_env_params.map_width + 2 * vision_power_map_padding,
+            ),
             dtype=jnp.int16,
         )
 
         # Update sensor mask based on the sensor range
+        max_sensor_range = env_params_ranges["unit_sensor_range"][-1]
```
Comment (CJBoey) [inline src/luxai_s3/env.py:123]:
```suggestion

```

File: src/luxai_s3/env.py [hunk 3]
```diff
@@ -9,79 +9,144 @@
...
+                slice_sizes=(
+                    max_sensor_range * 2 + 1,
+                    max_sensor_range * 2 + 1,
+                ),
+            )
             update = jnp.zeros_like(existing_vision_power)
-            for i in range(params.unit_sensor_range + 1):
-                update = update.at[i:params.unit_sensor_range * 2 + 1 - i, i:params.unit_sensor_range * 2 + 1 - i].set(i + 1)
+            for i in range(max_sensor_range + 1):
+                val = jnp.where(i > max_sensor_range - params.unit_sensor_range, i + 1 - params.unit_sensor_range, 0).astype(jnp.int16)
```
Comment (CJBoey) [inline src/luxai_s3/env.py:139]:
```suggestion
                val = jnp.where(i > max_sensor_range - params.unit_sensor_range - 1, i + 1 - params.unit_sensor_range, 0).astype(jnp.int16)
```

**Review by CJBoey — COMMENTED** [review]
Batching with different EnvParams works now but sensor masks no longer computed correctly, I tested the changes below and it seems to fix the issue.
image

**Comment by StoneT2000** [comment]
thanks for the suggestions, ill take a look @CJBoey
